### DISCUSSION TITLE
End to End Type Safety For InertiaJS Apps

### DISCUSSION THREAD

**Discussion by nk2580** [inertiajs/inertia#1789] [Ideas]
i've been playing around with Inertia for quite a while in PHP and i am keen to integrate it with a Go API however one thing i've not enjoyed is that the client side of Inertia doesn't have end to end type safety.

i would love to entertain the idea of how we might implement end to end type safety for client side applications, situations such as Date serialisation, and optional fields in responses are kind of painful to deal with when rapidly prototyping applications in Inertia.

i've used tools such as ConnectRPC in other projects with great success however the benefit of Inertia's backend focused cognitive model is just awesome, and i would love to see this become a norm for frameworks in PHP.

**Comment by riencoertjens** [comment]
have you seen this? haven't had a chance to try it out yet but looks like it might be what you need https://github.com/7nohe/laravel-typegen

**Comment by AlexVanderbist** [comment]
I can't recommend our Laravel Data & TypeScript transformer enough for this: https://spatie.be/docs/laravel-data/v4/advanced-usage/typescript
